### PR TITLE
Updated release workflow and removed the `ci` step from scripts

### LINKED ISSUE #199: Automated release not working
The release workflow is failing because it's missing a build step (`npm ci`). The build step can be added as a parameter to the `optic-release-automation-action` under `build-command`.

As part of this ticket we should remove the unnecessary `ci` script from the package.json (and replace the usage of it with appropriate scripts `lint` + `test`).

### PR DESCRIPTION
**Pull request by radomird**
Closes #199

### CODE REVIEW

**Comment by simoneb** [comment]
It's still not super clear to me why this is failing, I would like us to understand that first. 2 Q:

- does npm ci run our custom script or the built-in one?
- does the prepublishOnly script run before the action's custom build command or after?

**Comment by radomird** [comment]
> * does npm ci run our custom script or the built-in one?
`npm ci` is never run, only `npm run ci ` as part of the `prepublishOnly` script

> * does the prepublishOnly script run before the action's custom build command or after?
should run after it but there was no custom build command before

**Comment by radomird** [comment]
> > `npm ci` is never run, only `npm run ci ` as part of the `prepublishOnly` script
> 
> which fails, right? and it fails because it expects packages to be installed already, which they aren't

Correct!

> > should run after it but there was no custom build command before
> 
> okay but it should run after, and instead it's run before. maybe we need to change the action? the feeling I have is that we're running npm version before our custom build command perhaps. can you confirm if that's the case?

That is, unfortunately, also correct - `npm version` runs before the custom build commands.

**Comment by simoneb** [comment]
> That is, unfortunately, also correct - `npm version` runs before the custom build commands.

ok cool so we're on the same line. now we need to understand if there is a good reason why that is the case. I believe so, because in your custom build command you probably want to have the new version available already, in case you need to do anything with it. so what that leaves us with is, shall we change when the custom ci script in this package is called so we use a npm lifecycle script which is NOT executed when you run npm version?

I don't understand how the solution in this PR can work btw. If prepublishOnly is executed when running npm version, and if npm version is run before we install packages via npm ci in the custom build-command, this will keep failing pretty much in the same way. Am I missing anything?

**Comment by radomird** [comment]
> I don't understand how the solution in this PR can work btw. If prepublishOnly is executed when running npm version, and if npm version is run before we install packages via npm ci in the custom build-command, this will keep failing pretty much in the same way. Am I missing anything?

`prepublishOnly` will run when `npm publish` is executed, which is **after** the build commands. So I don't think we have an issue with running the `npm version` before the build commands.

**Comment by radomird** [comment]
Decided to rename the `ci` step to `test:ci` together with @simoneb.

**Review by simoneb — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -19,4 +19,6 @@ jobs:
           github-token: ${{ secrets.github_token }}
           npm-token: ${{ secrets[format('NPM_TOKEN_{0}', github.actor)] || secrets.NPM_TOKEN }}
           optic-token: ${{ secrets[format('OPTIC_TOKEN_{0}', github.actor)] || secrets.OPTIC_TOKEN }}
-          semver: ${{ github.event.inputs.semver }}
\ No newline at end of file
+          semver: ${{ github.event.inputs.semver }}
+          build-command: |
+            npm ci
```
[the remark this answers is not in the document]
Comment (radomird) [reply .github/workflows/release.yml:24]: I did fix this in: https://github.com/nearform/fast-jwt/pull/201 😄

**Pull request merged**